### PR TITLE
EPMRPP-38631 - React. Performance. WebPack re-configuration

### PR DESCRIPTION
**Pull request by mykhailo-kalashnikov-epam**

### CODE REVIEW

File: app/package.json [hunk 1]
```diff
@@ -10,27 +10,25 @@
...
     "test": "jest --notify",
     "test:watch": "jest --watch",
     "test:coverage": "jest --coverage",
     "build:lang:en": "webpack --display none --env.dev && babel localization/translator.js | node",
     "manage:translations": "webpack --display none --env.dev && babel-node ./localization/translatorRunner.js",
     "manage:translations:test": "webpack --display none --env.dev && babel-node ./localization/translatorRunnerCheck.js",
     "precommit": "lint-staged",
-    "postcommit": "git update-index --again"
+    "postcommit": "git update-index --again",
+    "storybook": "NODE_ENV=development start-storybook -p 9001 -c .storybook",
```
Comment (AmsterGet) [inline app/package.json]: Is it variable necessary to run storybook? It seems that `start-storybook` script doesn't use `NODE_ENV` environment variable

File: app/package.json [hunk 2]
```diff
@@ -10,27 +10,25 @@
     "format:js": "npm run eslint -- --fix",
     "format:scss": "npm run stylelint -- --fix",
     "format": "npm run format:js && npm run format:scss && prettier --write */**/*.md",
-    "dev": "webpack-dev-server --port 3000",
-    "build": "webpack -p --env.production",
-    "storybook": "start-storybook -p 9001 -c .storybook",
-    "storybook:build": "build-storybook -c .storybook -o .storybook_app",
+    "dev": "NODE_ENV=development webpack-dev-server --port 3000",
```
Comment (AmsterGet) [inline app/package.json]: I recommend to provide webpack `--env` option: `--env.NODE_ENV=development` and use it in the config file

File: app/package.json [hunk 3]
```diff
@@ -10,27 +10,25 @@
...
+    "build": "webpack -p",
     "test": "jest --notify",
     "test:watch": "jest --watch",
     "test:coverage": "jest --coverage",
     "build:lang:en": "webpack --display none --env.dev && babel localization/translator.js | node",
```
Comment (AmsterGet) [inline app/package.json]: Also please update this scripts with proper `--env` options

File: app/webpack.config.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+const merge = require('webpack-merge');
+const baseConfig = require('./webpack/base.config');
+const prodConfig = require('./webpack/prod.config');
+const devConfig = require('./webpack/dev.config');
+
+module.exports = () => {
+  const mode = process.env.NODE_ENV;
```
Comment (AmsterGet) [inline app/webpack.config.js]: With `--env.NODE_ENV=development` webpack option you will get access to `NODE_ENV` as a property of function parameters object

File: app/package.json (hunk 3 above)
Comment (mykhailo-kalashnikov-epam) [reply app/package.json]: production mode used by default.

**Pull request closed without merging**